test/modules/aws-cloudwatch-integration.ts

```diff
@@ -394,20 +397,151 @@ describe('AwsCloudwatch Integration Testing', () => {
+      `
+    SELECT *
+    FROM metric_alarm
+    WHERE alarm_name = '${alarmName}';
+  `,
+      (res: any[]) => expect(res.length).toBe(0),
+    ),
+  );
+
+  it('starts a transaction', begin());
+
+  it(
+    'adds a new alarm',
+    query(
+      `
+      INSERT INTO metric_alarm (alarm_name, alarm_description, actions_enabled, comparison_operator, datapoints_to_alarm, dimensions, metric_name, namespace, period, statistic, threshold, evaluation_periods)
+      VALUES ('${alarmName}', 'Metric alarm description', true, 'GreaterThanThreshold', 1, '[{"Name": "InstanceId", "Value": "test"}]', 'CPUUtilization', 'AWS/EC2', 60, 'Average', 10, 1);
+  `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+
+  it(
+    'check adds a new alarm',
+    query(
+      `
+    SELECT *
+    FROM metric_alarm
+    WHERE alarm_name = '${alarmName}';
+  `,
+      (res: any[]) => expect(res.length).toBe(1),
+    ),
+  );
+
+  it('applies the alarm change', commit());
+
+  it('starts a transaction', begin());
+
+  it(
+    'tries to update a metric alarm autogenerated field',
+    query(
+      `
+      UPDATE metric_alarm SET alarm_arn = '${alarmName}2' WHERE alarm_name = '${alarmName}';
+      `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+
+  it('applies the metric alarm change which will undo the change', commit());
+
+  it(
+    'check alarm ARN has not been updated',
+    query(
+      `
+    SELECT *
+    FROM metric_alarm
+    WHERE alarm_name = '${alarmName}' AND alarm_arn = '${alarmName}2';
+  `,
+      (res: any[]) => expect(res.length).toBe(0),
+    ),
+  );
+
+  it('starts a transaction', begin());
+
+  it(
+    'tries to add metric actions',
+    query(
+      `
+      UPDATE metric_alarm SET evaluation_periods=3, alarm_actions='${alarmActions}' WHERE alarm_name = '${alarmName}';
+      `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+
+  it('applies the metric alarm change to update periods', commit());
+
+  it(
+    'check evaluation periods has been modified',
+    query(
+      `
+    SELECT *
+    FROM metric_alarm
+    WHERE alarm_name = '${alarmName}' AND evaluation_periods=3;
+  `,
+      (res: any[]) => expect(res.length).toBe(1),
+    ),
+  );
+
   it('starts a transaction', begin());
 
   it(
-    'deletes all the log groups for the last time',
+    'deletes the metric alarm',
     query(
       `
-        DELETE FROM log_group;
+      DELETE FROM metric_alarm
+      WHERE alarm_name = '${alarmName}';
       `,
       undefined,
       true,
       () => ({ username, password }),
     ),
   );
 
+  it('applies the metric alarm delete', commit());
+
+  it(
+    'check deletes the metric alarm',
+    query(
+      `
+    SELECT *
+    FROM metric_alarm
+    WHERE alarm_name = '${alarmName}';
+  `,
+      (res: any[]) => expect(res.length).toBe(0),
+    ),
+  );
+
+  // cleanup
   it('applies deletion of all records', commit());
```

Comment:
It looks like this `commit` doesn't have the paired `begin`.

Not sure if you want to just remove it, or restore the `DELETE FROM log_group` transaction?